src/Compilers/Shared/CoreClrBuildClient.cs

```diff
@@ -46,14 +46,22 @@ protected override string GetSessionKey(BuildPaths buildPaths)
             return string.Empty;
         }
 
-        protected override async Task<BuildResponse> RunServerCompilation(List<string> arguments, BuildPaths buildPaths, string pipeName, string keepAlive, string libDirectory, CancellationToken cancellationToken)
+        protected override async Task<RunServerResult> RunServerCompilation(List<string> arguments, BuildPaths buildPaths, string pipeName, string keepAlive, string libDirectory, CancellationToken cancellationToken)
```

Comment:
How so?  There are two `return` paths here and all return a valid `RunServerResult` object.  That should always result in a completed `Task` value (second is in a blanket `catch` block).